script/generate_opengrep_checksums.sh

```diff
@@ -0,0 +1,201 @@
+    URL="${BASE_URL}/${VERSION}/${DIST}"
+    echo "  - ${DIST}..."
+
+    if curl -fsSL -o "${DIST}" "${URL}"; then
+        DOWNLOADED_COUNT=$((DOWNLOADED_COUNT + 1))
+
+        # Also download signature files if they exist
+        SIG_URL="${URL}.sig"
+        CERT_URL="${URL}.cert"
+
+        if curl -fsSL -o "${DIST}.sig" "${SIG_URL}" 2>/dev/null; then
+            echo "    ✓ Downloaded ${DIST}.sig"
+        fi
+
+        if curl -fsSL -o "${DIST}.cert" "${CERT_URL}" 2>/dev/null; then
+            echo "    ✓ Downloaded ${DIST}.cert"
+        fi
+    else
+        echo "    Warning: Failed to download ${DIST}"
+        echo "             (may not exist for this version)"
+        rm -f "${DIST}"
+    fi
+done
+
+if ((DOWNLOADED_COUNT == 0)); then
+    echo
+    echo "Error: No binaries could be downloaded. Check the version number."
+    exit 1
+fi
+
+echo
+echo "✓ Downloaded ${DOWNLOADED_COUNT}/${#PLATFORMS[@]} binaries"
+echo
+
+echo "Step 2: Generating checksums..."
+declare -A CHECKSUMS
+
+for DIST in opengrep_* *windows*.exe; do
+    if [[ -f "${DIST}" ]]; then
+        # Calculate SHA256 of the binary
+        if command -v sha256sum > /dev/null 2>&1; then
+            CHECKSUM=$(sha256sum "${DIST}" | cut -d' ' -f1)
+        elif command -v shasum > /dev/null 2>&1; then
+            CHECKSUM=$(shasum -a 256 "${DIST}" | cut -d' ' -f1)
+        else
+            echo "Error: Neither sha256sum nor shasum found"
+            exit 1
+        fi
+
+        CHECKSUMS["${DIST}"]="${CHECKSUM}"
+        echo "  ✓ ${DIST}: ${CHECKSUM}"
+
+        # Also checksum the signature file if it exists
+        if [[ -f "${DIST}.sig" ]]; then
+            if command -v sha256sum > /dev/null 2>&1; then
+                SIG_CHECKSUM=$(sha256sum "${DIST}.sig" | cut -d' ' -f1)
+            elif command -v shasum > /dev/null 2>&1; then
+                SIG_CHECKSUM=$(shasum -a 256 "${DIST}.sig" | cut -d' ' -f1)
+            fi
+            echo "    (signature: ${SIG_CHECKSUM})"
+        fi
+    fi
+done
+
+echo
+echo "Step 3: Updating ${INSTALL_SCRIPT}..."
+
+# Backup the original file
+cp "${INSTALL_SCRIPT}" "${INSTALL_SCRIPT}.bak"
+echo "  - Created backup: ${INSTALL_SCRIPT}.bak"
+
+# Update version
+"${SED_INPLACE[@]}" \
+  "s/^OPENGREP_VERSION = .*/OPENGREP_VERSION = '${VERSION}'/" \
+  "${INSTALL_SCRIPT}"
+echo "  ✓ Updated OPENGREP_VERSION to '${VERSION}'"
+
+# Update each checksum in BINARY_CHECKSUMS dictionary
+# Format: 'platform': (\n        'checksum'),
+for DIST in "${PLATFORMS[@]}"; do
+    if [[ -n "${CHECKSUMS[${DIST}]:-}" ]]; then
+        CHECKSUM="${CHECKSUMS[${DIST}]}"
+        # Use perl for multi-line replacement to maintain 80-char limit
+        # Match: 'dist': (\n        'old_checksum'),
+        # Replace with: 'dist': (\n        'new_checksum'),
+        perl -i -pe "
+          if (/'${DIST}': \(/) {
+            \$_ .= <>;
+            s/('${DIST}': \(\\n\\s+')[^']*('\\),)/\${1}${CHECKSUM}\${2}/;
+          }
```

Comment:
Perl is a heavy dependency to bring in for one line that could be achieved using existing dependencies. Plus, that should be fun to maintain.